components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodGatewayClientCustomizationProvider.kt

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package io.gitpod.jetbrains.remote
+
+import com.jetbrains.rdserver.unattendedHost.customization.GatewayClientCustomizationProvider
+import com.jetbrains.rdserver.unattendedHost.customization.controlCenter.DefaultGatewayControlCenterProvider
+import com.jetbrains.rdserver.unattendedHost.customization.controlCenter.GatewayControlCenterProvider
+import com.jetbrains.rdserver.unattendedHost.customization.controlCenter.GatewayHostnameDisplayKind
+import io.gitpod.jetbrains.remote.icons.GitpodIcons
+import javax.swing.Icon
+
+class GitpodGatewayClientCustomizationProvider : GatewayClientCustomizationProvider {
+    override val icon: Icon = GitpodIcons.Logo
+    override val title: String = "Gitpod Workspace"
+
+    override val controlCenter: GatewayControlCenterProvider = object : GatewayControlCenterProvider {
+        override fun getHostnameDisplayKind() = GatewayHostnameDisplayKind.ShowHostnameOnNavbar
+        override fun getHostnameShort() = if (System.getenv("GITPOD_WORKSPACE_ID") != null) {
```

Comment:
`System.getenv("GITPOD_WORKSPACE_ID") ?: "Gitpod"`

Ideally we should the completely gitpod workpace host which is fetched from supervisor info endpoint. We can set it in status endpoint as a custom env var which is starting the backend. Could you try it?

Reply:
Can you exemplify the "gitpod workpace host" stirng?
Is it like `gitpodio-gitpod-khhiqub2fgv.ws-us43xl.gitpod.io`?

Reply:
You can get host from https://github.com/gitpod-io/gitpod/blob/612041c5594e5d7b4c74fa58f3176ef3f478901f/components/supervisor-api/info.proto#L74
You can access info here https://github.com/gitpod-io/gitpod/blob/612041c5594e5d7b4c74fa58f3176ef3f478901f/components/ide/jetbrains/image/status/main.go#L69
And set as an env var here:  https://github.com/gitpod-io/gitpod/blob/612041c5594e5d7b4c74fa58f3176ef3f478901f/components/ide/jetbrains/image/status/main.go#L229

Something like `JETBRAINS_GITPOD_WORKSPACE_HOST`

Reply:
> Is it like gitpodio-gitpod-khhiqub2fgv.ws-us43xl.gitpod.io?

Note: This is exactly what JB IDE shows by default, when we hover the hostname on NavBar:
<img width="451" alt="image" src="https://user-images.githubusercontent.com/418083/166689904-0e35af60-4ff0-4c76-ac6c-1d1ba40e7fb3.png">

Also, if we need to build the url, there are already these ENV that allow us to build it:
```
GITPOD_WORKSPACE_ID=gitpodio-gitpod-khhiqub2fgv
GITPOD_WORKSPACE_URL=https://gitpodio-gitpod-khhiqub2fgv.ws-us43xl.gitpod.io
GITPOD_WORKSPACE_CLUSTER_HOST=ws-us43xl.gitpod.io
```

May I use them, or shall I proceed with `JETBRAINS_GITPOD_WORKSPACE_HOST` addition?

Reply:
Env var started with `GITPOD_` are internal details of supervisor and should not be used by clients. Values should be fetched via supervisor info endpoint.

Reply:
> This is exactly what JB IDE shows by default, when we hover the hostname on NavBar:

ok, then we don't need to customize it

Reply:
Ok, in any case we need to override the `GatewayClientCustomizationProvider` to be able to customize the logo.

So for the hostname I'll be relying on `DefaultGatewayControlCenterProvider().getHostnameShort()` which returns the hostname in the format we want (e.g `gitpodio-gitpod-khhiqub2fgv.ws-us43xl.gitpod.io`).

Code updated.

Reply:
actually we need to override host too, we are going to change host for ssh, so let's go with `JETBRAINS_GITPOD_WORKSPACE_HOST` to control it

Reply:
Follow-up issue created: https://github.com/gitpod-io/gitpod/issues/9814